### PR TITLE
Improve inspector sync performance

### LINKED ISSUE #988: AWS Inspector findings timeout
*Title*: *AWS Inspector findings timeout*

*Description*:
> Cartography is crashing when attempting to retrieve Inspector finding from ap-northeast-3

*To Reproduce*:
> cartography --experimental-neo4j-4x-support --neo4j-uri bolt://127.0.0.1:7687  --aws-sync-all-profiles

*Logs*:
> INFO:cartography.intel.aws.inspector:Syncing AWS Inspector findings for account XXX and region ap-northeast-3
INFO:backoff:Backing off inner_function(...) for 1.0s (botocore.exceptions.ClientError: An error occurred (InternalServerErrorException) when calling the ListFindings operation (reached max retries: 4): Endpoint request timed out)
WARNING:cartography.util:Backing off 1.0 seconds after 1 tries. Calling function <function aws_handle_regions.<locals>.inner_function at 0x106b0e710>
INFO:backoff:Backing off inner_function(...) for 0.2s (botocore.exceptions.ClientError: An error occurred (InternalServerErrorException) when calling the ListFindings operation (reached max retries: 4): Endpoint request timed out)
WARNING:cartography.util:Backing off 0.2 seconds after 2 tries. Calling function <function aws_handle_regions.<locals>.inner_function at 0x106b0e710>
INFO:backoff:Backing off inner_function(...) for 0.2s (botocore.exceptions.ClientError: An error occurred (InternalServerErrorException) when calling the ListFindings operation (reached max retries: 4): Endpoint request timed out)
WARNING:cartography.util:Backing off 0.2 seconds after 3 tries. Calling function <function aws_handle_regions.<locals>.inner_function at 0x106b0e710>
INFO:backoff:Backing off inner_function(...) for 6.4s (botocore.exceptions.ClientError: An error occurred (InternalServerErrorException) when calling the ListFindings operation (reached max retries: 4): Endpoint request timed out)
WARNING:cartography.util:Backing off 6.4 seconds after 4 tries. Calling function <function aws_handle_regions.<locals>.inner_function at 0x106b0e710>
ERROR:backoff:Giving up inner_function(...) after 5 tries (botocore.exceptions.ClientError: An error occurred (InternalServerErrorException) when calling the ListFindings operation (reached max retries: 4): Endpoint request timed out)

*Please complete the following information:*:
>  - Cartography 0.65.0

*Additional context*:
> Sounds like inspector does not exist in ap-northeast-3, so maybe that has something to do with it?

### PR DESCRIPTION
**Pull request by ramonpetgrave64**
The aws inspector api has some performance issues. This PR addresses those.

- The normal `SecurityAudit` aws policy does not have `inspector2` permissions, so we update our docs to mention that your aws identity must also include the `AmazonInspector2ReadOnlyAccess` policy.
- The inspector2 service is not available in the `ap-northeast-3` region, so we add `InternalServerErrorException` to `aws_handle_regions`, so we can bypass. Fixes #988 
- The inspector2 api will hang unless you specify some `filterCriteria`. Here we filter by both a list of known accounts, and by a finding's `status`.
- As far as we can tell, findings cannot be deleted, so if we were to fetch all findings, this is potentially millions of items. Instead, we only fetch findings that are either `OPEN` or `SUPPRESSED`, not `CLOSED`.
- I've also created a helper function `aws_paginate` to help with the boilerplate pagination code.
- The finding objects need to be loaded to Neo4j in batches, because they are a lot of data, and we don't want to exceed our databases' transaction size limit.
- 

I have some additional concerns about the `:HAS` relationship between findings and packages, but we will address that ina. future pr.

### CODE REVIEW

File: cartography/util.py [hunk 1]
```diff
@@ -133,6 +135,26 @@ def timed(*args, **kwargs):  # type: ignore
...
+    '''
+    Helper method for boilerplate boto3 pagination
+    The **kwargs will be forwarded to the paginator
+    '''
+    paginator = client.get_paginator(method_name)
+    items = []
+    i = 0
+    for i, page in enumerate(paginator.paginate(**kwargs), start=1):
+        if i % 100 == 0:
+            logger.info(f'fetching page number {i}')
```
Comment (heryxpc) [inline cartography/util.py:153]: 100 is the default page size used by paginator, could you pass that configuration specifically?
From: https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/inspector.html#paginators
Something like :
```
paginator.paginate(**kwargs, PaginationConfig={
        'PageSize': 100
    }
```

**Review by heryxpc — APPROVED** [review]
Added a general comment if a specific page size can be passed to `paginator.paginate` considering the page number is being logged based on the page size.

File: cartography/util.py (hunk 1 above)
Comment (ramonpetgrave64) [reply cartography/util.py:153]: Specifically for list_findings, it is 100. Other apis will have other defaults. And it's possible that some api's will error on a page size as high as 100.

It's my intention that this new `aws_pagiante` function be used for other sync modules, so I do not want to force the user to need to be aware of an optimal page size.

**Comment by ramonpetgrave64** [comment]
> Added a general comment if a specific page size can be passed to `paginator.paginate` considering the page number is being logged based on the page size.

I include the log for fetches that may have even thousands of pages, regardless of default page sizes. Only so we can know that cartography is not hanging.

**Comment by ramonpetgrave64** [comment]
Closing, reopened in #1004

**Pull request closed without merging**